wgsl/index.bs

```diff
@@ -632,16 +633,71 @@ module header and each usage of a GLSL method will add the appropriate
   </xmp>
 </div>
 
+## Module Constants ## {#module-constants}
+
+A *module constant* declares a name for a value, outside of all function declarations.
+The name is available for use after the end of the declaration,
+until the end of the WGSL program.
+
+When the declaration has no attributes, the value is the expression in the declaration.
+
+<div class='example' heading='Module constants'>
+  <xmp>
+    const golden : f32 = 1.61803398875;       # The golden ratio
+    const e2 : vec3<i32> = vec3<i32>(0,1,0);  # The second unit vector for three dimensions.
+  </xmp>
+</div>
+
+When the declaration uses the `constant_id` attribute,
+the constant is *pipeline-overridable*. In this case:
+
+  * The type must one of the [[#scalar-types]].
+  * The attribute's literal operand is known as the *pipeline constant ID*,
+    and must be a non-negative integer value representable in 32 bits.
+  * The application can specify its own value for the name at pipeline-creation time.
```

Comment:
What happens if the application provides a value for an ID which doesn't exist in the shader?

Reply:
I would say that's an application error.  I would lean toward the requiring the pipeline creation API to fail.
The downside to that is it may cause more bookkeeping in the API.

Reply:
Added this as an issue